reactive/apache_bigtop_base.py

```diff
@@ -45,3 +48,9 @@ def set_java_home():
         utils.re_edit_in_place('/etc/environment', {
             r'#? *JAVA_HOME *=.*': 'JAVA_HOME={}'.format(java_home),
         }, append_non_matches=True)
+
+        # If we've potentially setup services with the previous
+        # version of Java, set a flag that a layer can use to trigger
+        # a restart of those services.
+        if is_state('bigtop.available'):
+            set_state('java.changed')
```

Comment:
This should probably be `bigtop.java.changed` just so it doesn't potentially conflict with the java interface layer.

Otherwise, this looks good.